src/Microsoft.DotNet.Internal.Health/AzureTableHealthReportProvider.cs

```diff
@@ -4,59 +4,41 @@
 using System.Collections.Generic;
 using System.Linq;
-using System.Net;
-using System.Net.Http;
-using System.Net.Http.Headers;
 using System.Text.Json;
 using System.Text.Json.Serialization;
 using System.Threading.Tasks;
-using Microsoft.DotNet.Services.Utility;
+using Azure.Data.Tables;
+using Azure.Identity;
 using Microsoft.Extensions.Logging;
 using Microsoft.Extensions.Options;
 
 namespace Microsoft.DotNet.Internal.Health;
 
-public sealed class AzureTableHealthReportProvider : IHealthReportProvider, IDisposable
+public sealed class AzureTableHealthReportProvider : IHealthReportProvider
 {
-    private readonly HttpClient _client;
-    private readonly IOptionsMonitor<AzureTableHealthReportingOptions> _options;
     private readonly ILogger<AzureTableHealthReportProvider> _logger;
-    private readonly ExponentialRetry _retry;
-        
-    private static readonly JsonSerializerOptions s_jsonSerializerOptions = new JsonSerializerOptions {DefaultIgnoreCondition = JsonIgnoreCondition.WhenWritingNull};
+    private readonly TableClient _tableClient;
 
     public AzureTableHealthReportProvider(
         IOptionsMonitor<AzureTableHealthReportingOptions> options,
-        ILogger<AzureTableHealthReportProvider> logger,
-        IHttpClientFactory clientFactory,
-        ExponentialRetry retry)
+        ILogger<AzureTableHealthReportProvider> logger)
     {
-        _options = options;
         _logger = logger;
-        _retry = retry;
-        if (string.IsNullOrEmpty(options.CurrentValue.WriteSasUri))
+
+        if (string.IsNullOrEmpty(options.CurrentValue.ConnectionString))
         {
-            _logger.LogWarning("AzureTableHealth WriteSas is not configured, no status will be written to table");
+            throw new ArgumentException($"{nameof(AzureTableHealthReportingOptions.ConnectionString)} missing in {AzureTableHealthReportingOptions.HealthReportSettingsSection} section of app settings");
         }
-        _client = clientFactory.CreateClient();
-        _client.DefaultRequestHeaders.Add("x-ms-version", "2013-08-15");
-    }
-
-    private bool TryGetUrlParts(out string baseUrl, out string sasQuery)
-    {
-        AzureTableHealthReportingOptions options = _options.CurrentValue;
-        if (string.IsNullOrEmpty(options.WriteSasUri))
+        if (string.IsNullOrEmpty(options.CurrentValue.TableName))
         {
-            baseUrl = null;
-            sasQuery = null;
-            return false;
+            throw new ArgumentException($"{nameof(AzureTableHealthReportingOptions.TableName)} missing in {AzureTableHealthReportingOptions.HealthReportSettingsSection} section of app settings");
         }
 
-        var builder = new UriBuilder(options.WriteSasUri);
-        sasQuery = builder.Query;
-        builder.Query = null;
-        baseUrl = builder.ToString();
-        return true;
+        DefaultAzureCredential credential = string.IsNullOrEmpty(options.CurrentValue.ManagedIdentityClientId)
+            ? new()
```

Comment:
what authentication is used if there is no client id?

Reply:
System assigned managed identity. You only specify the client ID for user assigned ones

Reply:
but also to run it locally we need it this way :/ so probably is the right thing to do

Reply:
Yup, we need the empty option for local development, and also the system assigned managed identities, since I don't think they have a client ID. The docs always say you shouldn't put anything when you're using them